framework/db/SchemaBuilder.php

```diff
@@ -0,0 +1,412 @@
+     * will be implicitly called when an unknown method is being invoked.
+     * @param string $name the method name
+     * @param array $arguments method parameters
+     * @return mixed the method return value
+     */
+    public function __call($name, $arguments)
+    {
+        if ($name === 'default') {
+            return call_user_func_array([$this, '_default'], $arguments);
+        }
+
+        return parent::__call($name, $arguments);
+    }
+
+    /**
+     * Specify check value for the column
+     *
+     * @param string $check
+     * @return SchemaBuilder
+     */
+    public function check($check)
+    {
+        $this->check = $check;
+
+        return $this;
+    }
+
+    /**
+     * Build full string for create the column's schema
+     *
+     * @return string
+     */
+    public function __toString()
+    {
+        return
+            $this->schema .
+            $this->getLengthString() .
+            $this->getNullString() .
+            $this->getDefaultString() .
+            $this->getCheckString();
+    }
+
+    /**
+     * Specify default value for the column
+     *
+     * @param mixed $default
+     * @return SchemaBuilder
+     */
+    protected function _default($default = null)
+    {
+        $this->default = $default;
+
+        return $this;
+    }
+
+    /**
+     * Returns string with length of column
+     *
+     * @return string
+     */
+    protected function getLengthString()
+    {
+        return ($this->length !== null ? "({$this->length})" : '');
+    }
+
+    /**
+     * Returns string with NOT NULL if isNotNull is true, otherwise returns
+     * empty string
+     *
+     * @return string
+     */
+    protected function getNullString()
+    {
+        return ($this->isNotNull === true ? ' NOT NULL' : '');
+    }
+
+    /**
+     * Returns string with default value of column
+     *
+     * @return string
+     */
+    protected function getDefaultString()
+    {
+        $string = '';
+
+        if ($this->default !== null) {
+            $string .= ' DEFAULT ';
+            switch (gettype($this->default)) {
+                case 'integer':
+                case 'double':
+                    $string .= $this->default;
+                    break;
+                case 'boolean':
+                    $string .= $this->default ? 'TRUE' : 'FALSE';
+                    break;
+                default:
+                    $string .= "'{$this->default}'";
+            }
+        }
+
+        return $string;
+    }
+
+    /**
+     * Returns check value of column
+     *
+     * @return string
+     */
+    protected function getCheckString()
+    {
+        return ($this->check !== null ? " CHECK ({$this->check})" : '');
+    }
+
+    /**
+     * Create schema builder for types with length
+     *
+     * @param string $type schema of column
+     * @param integer $length length of column
+     * @return SchemaBuilder
+     */
+    protected static function createDefault($type, $length = null)
```

Comment:
maybe just `create`? "default" is already the default value of the colum so a bit confusing.